### PR TITLE
log exception correctly in DatasourceInputFormat and IndexGeneratorJob

### PR DESCRIPTION
**Pull request by guobingkun**
Exceptions in IndexGeneratorJob and DatasourceInputFormat were logged properly.

### CODE REVIEW

File: indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java [hunk 1]
```diff
@@ -160,7 +161,8 @@ private DatasourceInputSplit toDataSourceSplit(
       locations = getFrequentLocations(segments, fio, conf);
     }
     catch (Exception e) {
-      logger.error("Exception thrown finding location of splits", e);
+      logger.error(e, "Exception thrown finding location of splits");
+      throw Throwables.propagate(e);
```
Comment (himanshug) [inline indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java]: i don't this needs to be propagated. finding locations should only be best effort.

Comment (guobingkun) [inline indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputFormat.java]: @himanshug done.

**Comment by drcrallen** [comment]
:+1:

**Comment by himanshug** [comment]
:+1:

**Comment by drcrallen** [comment]
I filed https://github.com/metamx/java-util/pull/44 to hopefully make this less of an issue sometime in the future

**Pull request merged**